lib/ide/src/double_representation.rs

```diff
@@ -0,0 +1,79 @@
+//! A module with all functions used to synchronize different representations of our language
+//! module.
+
+use ast::IdMap;
+use data::text::Size;
+use data::text::Span;
+
+
+
+// ================
+// === Text API ===
+// ================
+
+/// Update IdMap to reflect the recent code change.
```

Comment:
If this is minimal, provisional algorithm, I'd like to see a TODO note with information what should be expected to work and what roughly is remaining.